src/Elliptic/Systems/ScalarGaussBonnet/Equations.cpp

```diff
@@ -0,0 +1,147 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Elliptic/Systems/ScalarGaussBonnet/Equations.hpp"
+
+#include <cstddef>
+
+#include "DataStructures/DataBox/PrefixHelpers.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/EagerMath/DotProduct.hpp"
+#include "DataStructures/Tensor/EagerMath/RaiseOrLowerIndex.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/MakeWithValue.hpp"
+
+namespace sgb {
+
+void curved_fluxes(const gsl::not_null<tnsr::I<DataVector, 3>*> flux_for_field,
+                   const tnsr::II<DataVector, 3>& inv_conformal_metric,
+                   const tnsr::I<DataVector, 3>& shift,
+                   const Scalar<DataVector>& lapse,
+                   const Scalar<DataVector>& conformal_factor,
+                   const tnsr::i<DataVector, 3>& field_gradient) {
+  raise_or_lower_index(flux_for_field, field_gradient, inv_conformal_metric);
+
+  for (size_t i = 0; i < 3; i++) {
+    flux_for_field->get(i) /= pow<4>(get(conformal_factor));
+    flux_for_field->get(i) -= get(dot_product(shift, field_gradient)) *
+                              shift.get(i) / square(get(lapse));
+  }
+}
+
+void face_fluxes(gsl::not_null<tnsr::I<DataVector, 3>*> flux_for_field,
+                 const tnsr::II<DataVector, 3>& inv_conformal_metric,
+                 const tnsr::I<DataVector, 3>& shift,
+                 const Scalar<DataVector>& lapse,
+                 const Scalar<DataVector>& conformal_factor,
+                 const tnsr::i<DataVector, 3>& face_normal,
+                 const Scalar<DataVector>& field) {
+  raise_or_lower_index(flux_for_field, face_normal, inv_conformal_metric);
+
+  for (size_t i = 0; i < 3; i++) {
+    flux_for_field->get(i) /= pow<4>(get(conformal_factor));
+    flux_for_field->get(i) -= get(dot_product(shift, face_normal)) *
```

Comment:
Same here. Also add a note that this whole term could actually be precomputed once at the start of the simulation as a possible performance optimization.